Weater-app/src/components/main/detailsDailyWeather/DetailsDailyWeather.jsx

```diff
@@ -9,55 +10,63 @@ import precipitationIcon from '../../../assets/image/precipitationIcon.png';
 import windIcon from '../../../assets/image/windIcon.png';
 import cloud from '../../../assets/image/cloud.png';
 
-import { MainDetailsDailyWeatherWrapper } from './styledComponents/DetailsDailyWeather';
+import {
+  blackTheme,
+  lightTheme,
+  MainDetailsDailyWeatherWrapper,
+} from './styledComponents/DetailsDailyWeather';
 
 const DetailsDailyWeather = () => {
   const state = useSelector((state) => state.app);
-  const { weather, isLoading, lang, theme } = state;
+  const { weather, isLoading, lang, isBlackTheme, isError } = state;
 
   return (
-    <MainDetailsDailyWeatherWrapper changeTheme={theme}>
-      {isLoading ? (
-        <h2>{languegeObject[lang].mainePageWeather.loading}</h2>
-      ) : (
-        <div>
-          <div className="detailsDailyItems">
-            <div className="imageCircle">
-              <img src={tempIcon} alt="tempIcon" />
+    <ThemeProvider theme={isBlackTheme ? blackTheme : lightTheme}>
+      <MainDetailsDailyWeatherWrapper>
+        {isLoading ? (
+          <h2>{languegeObject[lang].mainePageWeather.loading}</h2>
+        ) : isError ? (
+          <h2>{languegeObject[lang].mainePageWeather.error}</h2>
+        ) : (
+          <div>
+            <div className="detailsDailyItems">
+              <div className="imageCircle">
+                <img src={tempIcon} alt="tempIcon" />
+              </div>
+              <p>{languegeObject[lang].mainePageWeather.temperature}: </p>
+              <span>
+                {Math.floor(weather?.main.temp)}
+                <sup>&deg;</sup> {languegeObject[lang].mainePageWeather.feel}{' '}
+                {Math.floor(weather?.main.feels_like)}
+                <sup>&deg;</sup>
+              </span>
             </div>
-            <p>{languegeObject[lang].mainePageWeather.temperature}: </p>
-            <span>
-              {Math.floor(weather.main.temp && weather?.main.temp)}
-              <sup>&deg;</sup> {languegeObject[lang].mainePageWeather.feel}{' '}
-              {Math.floor(weather?.main.feels_like)}
-              <sup>&deg;</sup>
-            </span>
-          </div>
-          <div className="detailsDailyItems">
-            <div className="imageCircle">
-              <img src={precipitationIcon} alt="precipitationIcon" />
+            <div className="detailsDailyItems">
+              <div className="imageCircle">
+                <img src={precipitationIcon} alt="precipitationIcon" />
+              </div>
+              <p>{languegeObject[lang].mainePageWeather.precipitation}: </p>
+              <span>{weather?.weather[0].description}</span>
             </div>
-            <p>{languegeObject[lang].mainePageWeather.precipitation}: </p>
-            <span>{weather?.weather[0].description}</span>
-          </div>
-          <div className="detailsDailyItems">
-            <div className="imageCircle">
-              <img src={humidityIcon} alt="humidityIcon" />
+            <div className="detailsDailyItems">
```

Comment:
this block looks similar to previous one. can we move it to function and re-use it ?